spec/components/topic_query_spec.rb

```diff
@@ -268,33 +268,100 @@
       let!(:subcategory) { Fabricate(:category_with_definition, parent_category_id: category.id) }
       let(:subsubcategory) { Fabricate(:category_with_definition, parent_category_id: subcategory.id) }
 
+      # Not used in assertions but fabricated to ensure we're not leaking topics
+      # across categories
+      let!(:_category) { Fabricate(:category_with_definition) }
+      let!(:_subcategory) { Fabricate(:category_with_definition, parent_category_id: _category.id) }
```

Comment:
Can't `fab!` here because a nested category cannot be created unless `SiteSetting.max_category_nesting` is set.

Reply:
You could set the site setting in `fab!`, but this is fine.